discovery.md

```diff
@@ -521,14 +545,184 @@ When encoded in JSON, the response format MUST adhere to the following:
 }
 ```
 
-##### `/services?name={name}`
+Note: in the case of `406 Not Acceptable` or `409 Conflict`, it is an
+implementation choice as to how long these responses will be returned. But it
+is expected that a `404 Not Found` will eventually be returned.
```

Comment:
I'm not sure I understand what is meant here. Does this mean that, as a side effect of a `GET` request, a `Service` may be created? And if so, that process may fail, so subsequent requests for the same service will eventually return `404 Not Found`? I suspect that reading is incorrect because it seems wrong to me that anything would get created as a side effect here. But if that's the case, I don't understand the `406` and `409` return codes.  

Alternatively, it could be that the `Service` is being created as a result of a `POST` request as shown below. If so, this would mean that the Discovery API must maintain some sort of state around the creation of services that can be shared across requests. In all, this is confusing.

Update: it's becoming clearer as I continue reading. But I think it might be worth calling out that these responses can only exist in the case where a `POST` or `PUT` has recently been attempted with a previous request.

Reply:
I tried to add more text to help clarify why we're talking about this stuff... see if that helps.

Reply:
It does - thanks.